scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1433,9 +1452,13 @@
 
 (defn delete-pod
   "Kill this kubernetes pod. This is the same as deleting it."
-  [^ApiClient api-client compute-cluster-name ^V1Pod pod]
+  [^ApiClient api-client compute-cluster-name ^V1Pod pod & {:keys [grace-period-seconds]}]
   (let [api (CoreV1Api. api-client)
-        ^V1DeleteOptions deleteOptions (-> (V1DeleteOptionsBuilder.) (.withPropagationPolicy "Background") .build)
+        delete-options-builder
+        (cond-> (V1DeleteOptionsBuilder.)
+                true (.withPropagationPolicy "Background")
```

Comment:
May be worth a question. Do we want to delete it with background propagation policy, or should hard deletes have another policy?

Reply:
Sounds like it doesn't matter because we're using naked pods, so there are no child resources